internal/test/issues/issue-1039/issue_test.go

```diff
@@ -0,0 +1,209 @@
+package issue1039
+
+import (
+	_ "embed"
+	"encoding/json"
+	"github.com/oapi-codegen/nullable"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func ptr[T any](v T) *T {
+	return &v
+}
+
+func TestNullableTypesMarshal(t *testing.T) {
+	// include all fields in patch request
+	patchReq := PatchRequest{
+		ComplexOptionalNullable: nullable.Nullable[ComplexOptionalNullable]{
```

Comment:
can the order of fields in this function mirror what's in the other file? Will make it easier to diff